[BUG] kube-ovn-controller panic: assignment to entry in nil map

### Kube-OVN Version

v1.14.4

### Kubernetes Version

-

### Operation-system/Kernel Version

-

### Description

-

### Steps To Reproduce

-

### Current Behavior

```log
E0822 18:11:04.422447       7 runtime_faststr_swiss.go:265] "Observed a panic" panic="assignment to entry in nil map" panicGoValue="\"assignment to entry in nil map\"" stacktrace=<
        goroutine 1431 [running]:
        k8s.io/apimachinery/pkg/util/runtime.logPanic({0x55b8171b1350, 0x55b819129640}, {0x55b816c39420, 0x55b818b8be60})
                k8s.io/apimachinery@v0.33.1/pkg/util/runtime/runtime.go:107 +0xbc
        k8s.io/apimachinery/pkg/util/runtime.handleCrash({0x55b8171b1350, 0x55b819129640}, {0x55b816c39420, 0x55b818b8be60}, {0x55b819129640, 0x0, 0x10055b815c1efe4?})
                k8s.io/apimachinery@v0.33.1/pkg/util/runtime/runtime.go:82 +0x5a
        k8s.io/apimachinery/pkg/util/runtime.HandleCrash({0x0, 0x0, 0xc002fd56c0?})
                k8s.io/apimachinery@v0.33.1/pkg/util/runtime/runtime.go:59 +0x105
        panic({0x55b816c39420?, 0x55b818b8be60?})
                runtime/panic.go:792 +0x132
        github.com/kubeovn/kube-ovn/pkg/controller.(*Controller).checkServiceLBIPBelongToSubnet(0xc00065f508, 0xc004440008)
                github.com/kubeovn/kube-ovn/pkg/controller/service.go:550 +0x1bd
        github.com/kubeovn/kube-ovn/pkg/controller.(*Controller).handleUpdateService(0xc00065f508, 0xc006abd880)
                github.com/kubeovn/kube-ovn/pkg/controller/service.go:341 +0xdbe
        github.com/kubeovn/kube-ovn/pkg/controller.processNextWorkItem[...].func1(0xc0009a0c40, {0x55b815cbf854, 0xe?}, 0x55b817168238, 0x55b817178700, 0xc006abd880)
                github.com/kubeovn/kube-ovn/pkg/controller/controller.go:1434 +0xa2
        github.com/kubeovn/kube-ovn/pkg/controller.processNextWorkItem[...]({0x55b815cbf854, 0xe}, {0x55b8171cd070, 0xc000b61e60}, 0xc005a3a100, 0x55b817168238)
                github.com/kubeovn/kube-ovn/pkg/controller/controller.go:1440 +0x74
        github.com/kubeovn/kube-ovn/pkg/controller.(*Controller).startWorkers.runWorker[...].func49()
                github.com/kubeovn/kube-ovn/pkg/controller/controller.go:1470 +0x73
        k8s.io/apimachinery/pkg/util/wait.BackoffUntil.func1(0xc000781808?)
                k8s.io/apimachinery@v0.33.1/pkg/util/wait/backoff.go:226 +0x33
        k8s.io/apimachinery/pkg/util/wait.BackoffUntil(0xc004c8e580, {0x55b817179dc0, 0xc007389080}, 0x1, 0xc000538a80)
                k8s.io/apimachinery@v0.33.1/pkg/util/wait/backoff.go:227 +0xaf
        k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc004c8e580, 0x3b9aca00, 0x0, 0x1, 0xc000538a80)
                k8s.io/apimachinery@v0.33.1/pkg/util/wait/backoff.go:204 +0x7f
        k8s.io/apimachinery/pkg/util/wait.Until(...)
                k8s.io/apimachinery@v0.33.1/pkg/util/wait/backoff.go:161
        created by github.com/kubeovn/kube-ovn/pkg/controller.(*Controller).startWorkers in goroutine 111
                github.com/kubeovn/kube-ovn/pkg/controller/controller.go:1253 +0x3f13
 >
panic: assignment to entry in nil map [recovered]
        panic: assignment to entry in nil map
```

### Expected Behavior

-

--- a/pkg/controller/ip.go
+++ b/pkg/controller/ip.go
@@ -117,7 +117,9 @@ func (c *Controller) handleAddReservedIP(key string) error {
 		klog.Error(err)
 		return err
 	}
+	ip = ip.DeepCopy()
 	klog.V(3).Infof("handle add reserved ip %s", ip.Name)
+
 	if !ip.DeletionTimestamp.IsZero() {
 		klog.Infof("handle add process stop for deleting ip %s", ip.Name)
 		return nil
@@ -127,6 +129,7 @@ func (c *Controller) handleAddReservedIP(key string) error {
 		klog.Error(err)
 		return err
 	}
+
 	if ip.Spec.PodType != "" && ip.Spec.PodType != util.VM && ip.Spec.PodType != util.StatefulSet {
 		err := fmt.Errorf("podType %s is not supported", ip.Spec.PodType)
 		klog.Error(err)
@@ -177,16 +180,10 @@ func (c *Controller) handleAddReservedIP(key string) error {
 		klog.Error(err)
 		return err
 	}
+	if ip.Labels == nil {
+		ip.Labels = map[string]string{}
+	}
 	if ip.Labels[util.IPReservedLabel] != "false" {
-		cachedIP, err := c.ipsLister.Get(key)
-		if err != nil {
-			if k8serrors.IsNotFound(err) {
-				return nil
-			}
-			klog.Error(err)
-			return err
-		}
-		ip = cachedIP.DeepCopy()
 		ip.Labels[util.IPReservedLabel] = "true"
 		patchPayloadTemplate := `[{ "op": "%s", "path": "/metadata/labels", "value": %s }]`
 		raw, err := json.Marshal(ip.Labels)

--- a/pkg/controller/service.go
+++ b/pkg/controller/service.go
@@ -555,6 +555,10 @@ func diffSvcPorts(oldPorts, newPorts []v1.ServicePort) (toDel []v1.ServicePort) 
 }
 
 func (c *Controller) checkServiceLBIPBelongToSubnet(svc *v1.Service) error {
+	svc = svc.DeepCopy()
+	if svc.Annotations == nil {
+		svc.Annotations = map[string]string{}
+	}
 	subnets, err := c.subnetsLister.List(labels.Everything())
 	if err != nil {
 		klog.Errorf("failed to list subnets: %v", err)

--- a/pkg/controller/switch_lb_rule.go
+++ b/pkg/controller/switch_lb_rule.go
@@ -330,6 +330,9 @@ func generateHeadlessService(slr *kubeovnv1.SwitchLBRule, oldSvc *corev1.Service
 
 	if oldSvc != nil {
 		newSvc = oldSvc.DeepCopy()
+		if newSvc.Annotations == nil {
+			newSvc.Annotations = map[string]string{}
+		}
 		newSvc.Name = name
 		newSvc.Namespace = slr.Spec.Namespace
 		newSvc.Annotations[util.SwitchLBRuleVipsAnnotation] = slr.Spec.Vip

--- a/pkg/controller/vip.go
+++ b/pkg/controller/vip.go
@@ -417,6 +417,9 @@ func (c *Controller) podReuseVip(vipName, portName string, keepVIP bool) error {
 		return err
 	}
 	vip := oriVip.DeepCopy()
+	if vip.Labels == nil {
+		vip.Labels = map[string]string{}
+	}
 	var op string
 
 	if vip.Labels[util.IPReservedLabel] != "" {

--- a/pkg/controller/vpc_dns.go
+++ b/pkg/controller/vpc_dns.go
@@ -402,6 +402,9 @@ func setVpcDNSRoute(dp *v1.Deployment, subnetGw string) {
 			if err != nil {
 				klog.Errorf("failed to marshal routes %+v: %v", routes, err)
 			} else {
+				if dp.Spec.Template.Annotations == nil {
+					dp.Spec.Template.Annotations = map[string]string{}
+				}
 				dp.Spec.Template.Annotations[fmt.Sprintf(util.RoutesAnnotationTemplate, nadProvider)] = string(buf)
 			}
 			break

--- a/pkg/daemon/handler.go
+++ b/pkg/daemon/handler.go
@@ -390,6 +390,9 @@ func (csh cniServerHandler) UpdateIPCR(podRequest request.CniRequest, subnet, ip
 			klog.Error(err)
 		} else if ipCR.Spec.NodeName != csh.Config.NodeName {
 			ipCR := ipCR.DeepCopy()
+			if ipCR.Labels == nil {
+				ipCR.Labels = map[string]string{}
+			}
 			ipCR.Spec.NodeName = csh.Config.NodeName
 			ipCR.Spec.AttachIPs = []string{}
 			ipCR.Labels[subnet] = ""

